src/Microsoft.DotNet.Build.Tasks/PackageFiles/packageresolve.targets

```diff
@@ -85,14 +87,18 @@
     <!-- We may have an indirect package reference that we want to replace with a project reference -->
     <ItemGroup>
       <!-- Intersect project-refs with package-refs -->
-      <_ReferenceFileNamesToRemove Include="@(Reference)" Condition="'@(_ResolvedProjectReferencePaths->'%(FileName)%(Extension)')' == '%(FileName)%(Extension)'" />
+      <_ReferenceFileNamesToRemove Include="@(Reference)" Condition="'$(_BuildAgainstPackages)' == 'true' and '@(ProjectReference->'%(FileName) - %(KeepProjectReference)')' == '%(FileName) - true'" />
+      <_ReferenceFileNamesToRemove Include="@(Reference)" Condition="'$(_BuildAgainstPackages)' != 'true' and '@(_ResolvedProjectReferencePaths->'%(FileName)%(Extension)')' == '%(FileName)%(Extension)'" />
```

Comment:
It was previously necessary to provide the inverse logic to what was previously here so that we could prevent copying intersecting project refs with package refs.  At a certain point, I changed the tact and added the "ExtractProjectReferences;BeforeResolveReferences" targets so that the project references are no longer persisted in the build targets and this code is no longer necessary.  I'll remove it, thanks for pointing this out.